### PR TITLE
docs: Document releases. Closes #5379

### LINKED ISSUE #5379: TOB-ARGO-034: document release cycle
Examples:

* https://kubernetes.io/docs/setup/release/version-skew-policy/
* https://redis.io/topics/releases
* https://www.kernel.org/category/releases.html

### PR DESCRIPTION
**Pull request by alexec**
image

### CODE REVIEW

File: docs/releases.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Releases
+
+## Supported Versions
+
+Versions are expressed as x.y.z, where x is the major version, y is the minor version, and z is the patch version,
```
Comment (simster7) [inline docs/releases.md:5]:
```suggestion
Versions are expressed as vX.Y.Z, where X is the major version, Y is the minor version, and Z is the patch version,
```

File: docs/releases.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+following Semantic Versioning terminology.
```
Comment (simster7) [inline docs/releases.md:6]: Link to the semver definition?

File: docs/releases.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+We maintain release branches for the most recent two minor releases.
+
+Fixes may be backported to release branches, depending on severity, risk, and, feasibility.
+
+If a release contains breaking changes, or CVE fixes, this will documented in the release notes.
+
+## Supported Version Skew
+
+Both the `argo-server` and `argocli` should be the same version as the controller.
```
Comment (simster7) [inline docs/releases.md:16]: The `argocli` and the `argo-server` are the same binary/image. Should we just say "`argocli` (which contains the Argo Server"?

File: docs/releases.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+# Release Cycle
+
+For **unstable**, we build and tag `latest` images for every commit to master.
+
+New major version are released roughly ever 3 months. Release candidates for each major release are typically available
+for 6 weeks before the release becomes generally available.
```
Comment (simster7) [inline docs/releases.md:23]:
```suggestion
New major versions are released roughly every 3 months. Release candidates for each major release are typically available
for 6 weeks before the release becomes generally available.
```

**Review by simster7 — DISMISSED** [review]

File: docs/releases.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Releases
+
+## Supported Versions
+
+Versions are expressed as x.y.z, where x is the major version, y is the minor version, and z is the patch version,
+following Semantic Versioning terminology.
+
+We maintain release branches for the most recent two minor releases.
+
+Fixes may be backported to release branches, depending on severity, risk, and, feasibility.
```
Comment (terrytangyuan) [inline docs/releases.md:10]: Are issues like https://github.com/argoproj/argo-workflows/issues/5326 for this purpose? I see maintainers have edited the issue description. Could we document that process as well? Is this a place where community members can chime in and see if there other fixes that should be backported?

File: docs/releases.md (hunk 1 above)
Comment (alexec) [reply docs/releases.md:5]: I copied this from Kubernetes

File: docs/releases.md (hunk 3 above)
Comment (alexec) [reply docs/releases.md:16]: I think most users do not know that.

File: docs/releases.md (hunk 5 above)
Comment (alexec) [reply docs/releases.md:10]: There is no formal process for this.

**Review by simster7 — APPROVED** [review]

**Pull request merged**